### PR TITLE
Book review

### PR DESCRIPTION
**Pull request by dungdm93**

### CODE REVIEW

File: app/views/reviews/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<%= form_for @review, remote: true,
+             html: {role: :form, 'data-model' => 'review'} do |f| %>
```
Comment (dieunb) [inline app/views/reviews/_form.html.erb]:
```
data-model:  'review'
```

File: app/views/reviews/_review.html.erb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            class: "rating", id: "book-rating" %>
+      </div>
+    </div>
+  </div>
+  <%# render @commnets %>
+</div>
+
+<!-- rating script -->
+<script>
+  $('#book-rating').rating({
```
Comment (dieunb) [inline app/views/reviews/_review.html.erb]: move js to application.js

File: app/views/reviews/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<% provide(:title, "Review: #{@review.book.title}") %>
```
Comment (dieunb) [inline app/views/reviews/show.html.erb]: remove `()`

File: config/routes.rb [hunk 4]
```diff
@@ -15,8 +15,14 @@
   resources :categories
   resources :books
   mount PdfjsViewer::Rails::Engine => "/pdfjs", as: 'pdfjs'
+
   resources :book_requests, only: [:create, :destroy, :index]
 
+  resources :reviews do
+    resources :comments,    only: [:create]
+  end
+  resource :comments,       only: [:destroy]
```
Comment (dieunb) [inline config/routes.rb]: remove

File: app/views/reviews/_form.html.erb (hunk 1 above)
Comment (dungdm93) [inline app/views/reviews/_form.html.erb]: ```data-model``` contain ```-``` character so we can't use ```data-model:  'review'```

File: app/views/reviews/_review.html.erb (hunk 2 above)
Comment (dungdm93) [inline app/views/reviews/_review.html.erb]: this js must be under ```hidden_field_tag```.
Whenever i move it to head, it don't work. :(

File: app/models/review.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  # has_many   :comments, dependent: :destroy
+
+  before_validation :normalize_attributes
+  validates :user,    presence: true
+  validates :book,    presence: true
+  validates :content, presence: true
+  validates :rate,    inclusion: {in: 1..100}
+
+  scope :user, -> (use) {where user: use}
+  scope :book, -> (boo) {where book: boo}
```
Comment (dieunb) [inline app/models/review.rb]: book

File: app/helpers/books_helper.rb [hunk 6]
```diff
@@ -5,8 +5,13 @@ def cover_url(book)
 
   def related_books(book)
     # placeholder for later functionality
-    Book.where(category: book.category)
+    return unless book
+    Book.category(book.category)
         .where.not(id: book)
```
Comment (dieunb) [inline app/helpers/books_helper.rb]: Không move scope luôn em.

**Pull request merged**